samples/boards/nrf/clock_skew/README.rst

```diff
@@ -0,0 +1,116 @@
+.. _nrf-clock-skew-sample:
+
+nRF5x Clock Skew Demo
+#####################
+
+Overview
+********
+
+This sample uses the API for correlating time sources to measure the
+skew between HFCLK (used for the CPU) and LFCLK (used for system time).
```

Comment:
Okay, makes sense, but this would inherently mean that measuring the skew with the discontinuous time source (for example, the HFCLK that shuts off during sleep), would only be able to be done before the time source 'becomes' discontinuous (e.g. when the SoC goes into sleep). 
Would it then still be possible / make sense to do the conversion (properly) later into the application? If not, that would kind of defeat the purpose for this whole API (for our use-case), right?

[I really hope you'll respond with 'oh this is not a problem, we can find a quick solution for this by adding xyz', because basically the whole use-case of the API for us is counting on this, although it makes a lot of sense that the time sources need to be continuous]

Reply:
How this affects your application depends on details I don't know.  Let's say you wake up once an hour, run for two seconds, and believe your RTC chip exactly matches external time.  Then on startup you could use the API we've been discussing elsewhere to advance the Zephyr tick counter by the number of ticks in 59 minute 58 seconds (or, better, the difference between current time and last went-to-sleep time).  Unless you're measuring to microsecond resolution no skew that occurs within that two seconds is relevant, and you can just use 1 as the skew, and not bother calculating or storing it anywhere.

My expectation, from non-formal experiments with Nordic-based boards that use an external crystal for the local reference that need to be validated by long-running experiences, is that an approximate skew can be provided by default.  E.g. on most Nordic-based boards it's about 25 ppm.  It's possible that the skew can be board-dependent, and certain that it can be affected by temperature and passage of time, so in critical applications you would want to recalibrate it once in a while.  Recalibration does mean that the clocks must be running for the entire interval between the two captures.

In my case I deploy two sensors in a location at 2021-03-01T00:00:00Z and come back several months later.  Sensor A detected an event at a time that it converts to 2021-04-01T01:23:30Z.  At 25 ppm error over the 31 days in March the local time would be off by as much as 67 seconds, so if sensor B detected an event at 2021-04-01T01:22:25Z I don't know whether they were the same event.  If I can get the error down to 2 ppm I know it's not, because the error is less than 6 seconds over 31 days.

Reply:
Hmm interesting. [thinking out loud] then the plan for us could be to use the API to:
- per board run calibration tests over longer interval (without power management, so no sleep)
- run the skew calculation at boot / start application (before sleep has occured) as semi-calibration (since after sleep it wouldn't be possible)
- compare both?

We don't have a (time) critical application, but since the use-case is to use it as the (elapsed time during sleep) time reference for the system timer, I would say that would make it pretty critical. 

It could definitely be that above solution combined with the other API we've been discussing could actually cover everything.